src/stratigraphy/groundwater/groundwater_extraction.py

```diff
@@ -0,0 +1,420 @@
+
+    depth: float  # Depth of the groundwater relative to the surface
+    date: str | None = (
+        None  # Date of the groundwater measurement, if several dates
+        # are present, the date of the document the last measurement is taken
+    )
+    elevation: float | None = None  # Elevation of the groundwater relative to the mean sea level
+    rect: fitz.Rect | None = None  # The rectangle that contains the extracted information
+    page: int | None = None  # The page number of the PDF document
+
+    def is_valid(self) -> bool:
+        """Checks if the information is valid.
+
+        Returns:
+            bool: True if the information is valid, otherwise False.
+        """
+        return self.depth > 0
+
+    def __str__(self) -> str:
+        """Converts the object to a string.
+
+        Returns:
+            str: The object as a string.
+        """
+        return (
+            f"GroundwaterInformation("
+            f"date={self.date}, "
+            f"depth={self.depth}, "
+            f"elevation={self.elevation}, "
+            f"page={self.page})"
+        )
+
+    def to_dict(self) -> dict:
+        """Converts the object to a dictionary.
+
+        Returns:
+            dict: The object as a dictionary.
+        """
+        return {
+            "date": self.date,
+            "depth": self.depth,
+            "elevation": self.elevation,
+            "page": self.page if self.page else None,
+            "rect": [self.rect.x0, self.rect.y0, self.rect.x1, self.rect.y1] if self.rect else None,
+        }
+
+    def compare_dates(self, date1, date2) -> bool:
+        """Compares two dates.
+
+        Args:
+            date1 (str): The first date.
+            date2 (str): The second date.
+
+        Returns:
+            bool: True if the dates are equal, otherwise False.
+        """
+        # Define possible date formats
+        date_formats = ["%Y-%m-%d", "%d.%m.%Y"]
+
+        # Try to parse date1
+        for fmt in date_formats:
+            try:
+                parsed_date1 = datetime.strptime(date1, fmt)
+                break
+            except ValueError:
+                continue
+            except TypeError:
+                logger.warning("Groundtruth or extracted date is None.")
+                continue
+        else:
+            return False  # date1 does not match any format
+
+        # Try to parse date2
+        for fmt in date_formats:
+            try:
+                parsed_date2 = datetime.strptime(date2, fmt)
+                break
+            except TypeError:
+                logger.warning("Groundtruth or extracted date is None.")
+                continue
+            except ValueError:
+                continue
+        else:
+            return False  # date2 does not match any format
+
+        # Compare the parsed dates
+        return parsed_date1 == parsed_date2
+
+    def is_extracted_information_correct(self, extracted_information: "GroundwaterInformation") -> bool:
```

Comment:
Essentially this is an "equals" method for GroundwaterInformation objects, correct?

Reply:
Yes

Reply:
I chose to overwrite the __eq__ operator instead to make it clearer.